### DISCUSSION TITLE
Convention of coordination

### DISCUSSION THREAD

**Discussion by llcc** [mtex-toolbox/mtex#56] [Ask Anything] [answered]
Hi, thanks for this great package for matlab.

I used it to generate my ebsd figure. However, it seems like that there are some problems about the coordination setting. I would like to generate the figure like:
![image](https://cloud.githubusercontent.com/assets/3996879/7430870/6578a73c-f053-11e4-9d95-1a577ae7e3d2.png)

But it looks like below by default.
![image](https://cloud.githubusercontent.com/assets/3996879/7430878/82123fd4-f053-11e4-8357-60a165d3c7d4.png)

I tried to change the coordination by `setMTEXpref('xAxisDirection','west'` and tried several options，
but nothing happened. So can you give me some help?

**Comment by ralfHielscher** [comment]
After you have plotted the data you can change the the direction of the x axis in the figure menu MTEX/x-axis direction

the command `setMTEXpref('xAxisDirection','west')` is correct but should be used before plotting. A shortcut would be

``` matlab
plotx2west
plotzOutOfPlane
plot(ebsd)
```

I hope this helps,
Ralf

**Comment by llcc** [comment]
Absolutely I can change the axis manually. But I have lots of datas, so I plan to loop them all and generate everything I need. I tried what you told me above, but still can not make it. So can i send you source data and the snippets, so that you can help me to find the point of this issue?

But it seems github has not yet supported .osc files? So can i ask you e-mail address to send you two files? Thanks in advance!

First the code snippets:

``` matlab
%% Import Script for EBSD Data
%
% This script was automatically created by the import wizard. You should
% run the whoole script or parts of it in order to import your data. There
% is no problem in making any changes to this script.

%% Specify Crystal and Specimen Symmetries

% crystal symmetry
CS = {... 
  crystalSymmetry('432', [2.87 2.87 2.87], 'mineral', 'Iron - Alpha', 'color', 'light blue'),...
  crystalSymmetry('321', [4.052 4.052 5.086], 'X||a', 'Y||b*', 'Z||c*', 'mineral', 'Iron Silicon', 'color', 'light green')};

% plotting convention
%% Specify File Names
plotx2west
plotzOutOfPlane
% path to files
pname = 'C:\Users\admin\Desktop\4N_Fe4wtSi';

% which files to be imported
fnames = dir([pname, '\1-1.osc']);
for i = 1:length(fnames)
    fname = [pname '\' fnames(i).name]
    %% Import the Data
    %  create an EBSD variable containing the data
    ebsd = loadEBSD(fname,CS,'interface','osc',...
    'convertEuler2SpatialReferenceFrame');
    ebsd = ebsd('Iron - Alpha')

    oM = ipdfHSVOrientationMapping(ebsd)
    h=plot(ebsd, oM.orientation2color(ebsd.orientations), 'figSize', 'large')
    fname = fname(1:end-4)
    hgexport(gcf, [fname '_ipf.jpg'], hgexport('factorystyle'), 'Format', 'jpeg');
    grains = calcGrains(ebsd, 'angle', 10*degree)
    export(grains.meanRotation, [fname '_ipf.txt'])
end

```

**Comment by ralfHielscher** [comment]
Hi llcc,

looking at you code everything seems to be ok. Maybe for testing you can try

``` matlab
plotx2west
plotzIntoPlane
plot([xvector,yvector],'labels',{'x','y'})
```

Ralf

**Comment by llcc** [comment]
@ralfHielscher Thanks for your help.
Finally i made it by changing the direction with 

``` matlab
plotx2east
plotzIntoPlane
```

I am not very clear about the internal mechanism of the direction. So can you give me a brief introduction about it or can you give me a link about it if there is. Thanks!

**Comment by ralfHielscher** [comment, marked as answer]
By using the commands

``` matlab
plotx2east
plotx2west
plotx2north
plotx2south
plotzIntoPlane
plotzOutOfPlane
```

all alignments of the reference frame should be possible to define. This is stored internally as mtexPref and can be accessed by

``` getMTEXPref```
All plotting routines read the value there and alligne the plot accordingly.

I hope this helps,
Ralf.


```
